test/unit-test/mqtt_agent_utest.c

```diff
@@ -1380,41 +1413,68 @@ void test_MQTTAgent_CommandLoop_with_eventCallback( void )
 
     MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
     MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
-    MQTT_ProcessLoop_StubWithCallback( MQTT_ProcessLoop_stub );
+    MQTT_ProcessLoop_Stub( MQTT_ProcessLoop_stub );
 
     mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
 
     TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
     /* Ensure that callback is invoked. */
-    TEST_ASSERT_EQUAL( 8, commandCompleteCallbackCount );
+    TEST_ASSERT_EQUAL( 1, commandCompleteCallbackCount );
     /* Ensure that acknowledgment is cleared. */
     TEST_ASSERT_EQUAL( 0, mqttAgentContext.pPendingAcks[ 0 ].packetId );
     TEST_ASSERT_EQUAL( NULL, mqttAgentContext.pPendingAcks[ 0 ].pOriginalCommand );
 
     /* Invoking mqttEventCallback with MQTT_PACKET_TYPE_PUBLISH packet type. */
     packetType = MQTT_PACKET_TYPE_PUBLISH;
+    commandCompleteCallbackCount = 0;
 
     MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
     MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
-    MQTT_ProcessLoop_StubWithCallback( MQTT_ProcessLoop_stub );
+    MQTT_ProcessLoop_Stub( MQTT_ProcessLoop_stub );
 
     mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
 
     TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
     /* Ensure that callback is invoked. */
-    TEST_ASSERT_EQUAL( 9, commandCompleteCallbackCount );
+    TEST_ASSERT_EQUAL( 1, commandCompleteCallbackCount );
     TEST_ASSERT_EQUAL( 1, publishCallbackCount );
 
     /* Test case when completion callback is NULL. */
     commandToSend.pCommandCompleteCallback = NULL;
+    commandCompleteCallbackCount = 0;
 
     MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
     MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
-    MQTT_ProcessLoop_StubWithCallback( MQTT_ProcessLoop_stub );
+    MQTT_ProcessLoop_Stub( MQTT_ProcessLoop_stub );
 
     mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
 
     TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
     /* Ensure that callback is not invoked. */
-    TEST_ASSERT_EQUAL( 9, commandCompleteCallbackCount );
+    TEST_ASSERT_EQUAL( 0, commandCompleteCallbackCount );
+}
+
+/**
+ * @brief Test MQTT_ProcessLoop failure case via calling it multiple times.
+ */
+void test_MQTT_ProcessLoop_failure( void )
```

Comment:
This name can be more descriptive as to what we're testing
```suggestion
void test_MQTTAgent_CommandLoop_ProcessLoop_failure( void )
```